spacy/ml/models/coref.py

```diff
@@ -0,0 +1,595 @@
+from typing import List, Tuple
+import torch
+
+from thinc.api import Model, chain, tuplify
+from thinc.api import PyTorchWrapper, ArgsKwargs
+from thinc.types import Floats2d, Ints1d, Ints2d
+from thinc.util import xp2torch, torch2xp
+
+from ...tokens import Doc
+from ...util import registry
+from .coref_util import add_dummy, get_sentence_ids
+
+
+@registry.architectures("spacy.Coref.v1")
+def build_wl_coref_model(
+    tok2vec: Model[List[Doc], List[Floats2d]],
+    embedding_size: int = 20,
+    hidden_size: int = 1024,
+    n_hidden_layers: int = 1,  # TODO rename to "depth"?
+    dropout: float = 0.3,
+    # pairs to keep per mention after rough scoring
+    # TODO change to meaningful name
+    rough_k: int = 50,
+    # TODO is this not a training loop setting?
+    a_scoring_batch_size: int = 512,
+    # span predictor embeddings
+    sp_embedding_size: int = 64,
+):
+    # TODO fix this
+    try:
+        dim = tok2vec.get_dim("nO")
+    except ValueError:
+        # happens with transformer listener
+        dim = 768
+
+    with Model.define_operators({">>": chain}):
+        # TODO chain tok2vec with these models
+        coref_scorer = PyTorchWrapper(
+            CorefScorer(
```

Comment:
Can we rename `CorefScorer` to something that doesn't imply it's an evaluation script (like our normal `scorer`) and that is less conflicting with the existing `coref_scorer.py`?

Reply:
Renamed to `CorefClusterer`, which I don't like much, but doesn't have the ambiguity issues. 

For the record, the terminology here probably came from the reference implementation. There are still some sub-components that are called "Scorers", and honestly they'd be harder to rename, since what they do is score candidates.